backend/accounting-service/src/main/kotlin/services/projects/v2/ProjectService.kt

```diff
@@ -1245,6 +1249,267 @@ class ProjectService(
+                    left join project.groups g on g.id = a.group_id
+                    where project_id = :project and now() < expires
+                    order by expires desc
+                """
+            ).rows.groupBy { it.getAs<UUID>("token") }.map { entry ->
+                ProjectInviteLink(
+                    entry.key.toString(),
+                    entry.value.first().getAs<OffsetDateTime>("expires").toEpochSecond(),
+                    entry.value.mapNotNull { it.getString("group_id") },
+                    ProjectRole.valueOf(entry.value.first().getString("role_assignment")!!)
+                )
+            }
+        }
+
+        return PageV2(20, result, null)
+    }
+
+    suspend fun deleteInviteLink(
+        actorAndProject: ActorAndProject,
+        request: ProjectsDeleteInviteLinkRequest,
+        ctx: DBContext = db
+    ) {
+        val project = actorAndProject.requireProject()
+
+        ctx.withSession { session ->
+            requireAdmin(actorAndProject.actor, listOf(project), session)
+
+            val success = session.sendPreparedStatement(
+                {
+                    setParameter("project", project)
+                    setParameter("token", request.token)
+                },
+                """
+                    delete from project.invite_links where project_id = :project and token = :token
+                """
+            ).rowsAffected > 0
+
+            if (!success) {
+                throw RPCException(
+                    "Unable to delete invitation link",
+                    HttpStatusCode.BadRequest
+                )
+            }
+        }
+    }
+
+    suspend fun updateInviteLinkRoleAssignment(
+        actorAndProject: ActorAndProject,
+        request: ProjectsUpdateInviteLinkRoleAssignmentRequest,
+        ctx: DBContext = db
+    ) {
+        val project = actorAndProject.requireProject()
+
+        if (!listOf(ProjectRole.ADMIN, ProjectRole.USER).contains(request.role)) {
+            throw RPCException("Role assignment can only be either Admin or User", HttpStatusCode.BadRequest)
+        }
+
+        ctx.withSession { session ->
+            requireAdmin(actorAndProject.actor, listOf(project), session)
+
+            val success = session.sendPreparedStatement(
+                {
+                    setParameter("project", project)
+                    setParameter("token", request.token)
+                    setParameter("role", request.role.name)
+                },
+                """
+                    update project.invite_links
+                    set role_assignment = :role
+                    where project_id = :project and token = :token
+                """
+            ).rowsAffected > 0
+
+            if (!success) {
+                throw RPCException(
+                    "Unable to update role assignment",
+                    HttpStatusCode.BadRequest
```

Comment:
A better error message should probably state that the link might no longer be valid/has been deleted.